Explorer/Assets/DCL/Audio/Systems/LandscapeAudioCullingSystem.cs

```diff
@@ -0,0 +1,307 @@
+            else { updateFramesCounter--; }
+        }
+
+        private void InitializeTerrainAudioStates()
+        {
+            IReadOnlyList<Terrain> terrains = terrainGenerator.GetTerrains();
+            landscapeAudioStates = new NativeArray<LandscapeAudioState>(terrains.Count, Allocator.Persistent);
+            landscapeAudioSourcesPositions = new NativeArray<NativeArray<int2>>(terrains.Count, Allocator.Persistent);
+            int halfTerrainChunkSize = terrainGenerator.GetChunkSize() / 2;
+            audioListeningDistanceThreshold = math.pow(halfTerrainChunkSize + landscapeAudioSystemSettings.ListeningDistanceThreshold, 2);
+            audioMutingDistanceThreshold = math.pow(halfTerrainChunkSize + landscapeAudioSystemSettings.MutingDistanceThreshold, 2);
+
+            for (var i = 0; i < terrains.Count; i++)
+            {
+                Terrain terrain = terrains[i];
+                Bounds bounds = GetTerrainBoundsInWorldSpace(terrain);
+                landscapeAudioSourcesPositions[i] = CalculateAudioSourcesPositions(terrain.terrainData, bounds);
+
+                landscapeAudioStates[i] = new LandscapeAudioState
+                {
+                    CenterOfTerrain = new float2(bounds.center.x, bounds.center.z),
+                };
+            }
+        }
+
+        private void InitializeOceanBounds()
+        {
+            Transform ocean = terrainGenerator.GetOcean();
+            MeshRenderer[] renderers = ocean.GetComponentsInChildren<MeshRenderer>();
+
+            // some water chunks are disabled on purpose, we dont want to re-enable them
+            var waterRenderers = renderers.Where(meshRenderer => meshRenderer.enabled).ToList();
+            oceanListeningDistanceThreshold = math.pow(landscapeAudioSystemSettings.OceanDistanceThreshold, 2);
+
+            oceanAudioStates = new NativeArray<OceanAudioState>(waterRenderers.Count, Allocator.Persistent);
+
+            for (var i = 0; i < waterRenderers.Count; i++)
+            {
+                MeshRenderer waterChunk = waterRenderers[i];
+                Bounds bounds = waterChunk.bounds;
+
+                oceanAudioStates[i] = new OceanAudioState
+                {
+                    Bounds = new Bounds
+                    {
+                        center = bounds.center,
+                        extents = bounds.extents + Vector3.one,
+                    },
+                };
+            }
+        }
+
+        private NativeArray<int2> CalculateAudioSourcesPositions(TerrainData terrainData, Bounds worldBounds)
+        {
+            int rowsPerChunk = landscapeAudioSystemSettings.RowsPerChunk;
+            int retryAttempts = landscapeAudioSystemSettings.AudioSourcePositioningRetryAttempts;
+            Vector3 terrainSize = terrainData.size;
+            int cellWidth = (int)terrainSize.x / rowsPerChunk;
+            int cellLength = (int)terrainSize.z / rowsPerChunk;
+
+            var positions = new NativeList<int2>(Allocator.Temp);
```

Comment:
You can initialize the list as persistent, then return it `AsArray`. Thus, you don't need to copy its data into array.

`positions.ToArray(Allocator.Persistent);` -> `positions.AsArray()`

Reply:
Great point!